File: test_parse_md.py
import re

with open('data/files/incoming/02a47a5f-8bf7-4260-9ab7-a00caf9ee0fb_template_structured.md', 'r') as f:
    text = f.read()

activeChapter = "一、投标函"
cleanActiveChapter = re.sub(r'^第[一二三四五六七八九十百]+章', '', activeChapter).strip()

lines = text.split('\n')
capturing = False
captureLevel = 0
result = []

for line in lines:
    headingMatch = re.match(r'^(#{1,6})\s+(.*)', line)
    
    if headingMatch:
        level = len(headingMatch.group(1))
        title = headingMatch.group(2).strip()
        
        if not capturing:
            if activeChapter in title or (cleanActiveChapter and cleanActiveChapter in title):
                capturing = True
                captureLevel = level
                result.append(line)
        else:
            if level <= captureLevel:
                break
            else:
                result.append(line)
    elif capturing:
        result.append(line)

print("length:", len(result))
print("\n".join(result[:10]))

File: test_parse_md.js
const fs = require('fs');

const data = {
  original_markdown: fs.readFileSync('data/files/incoming/02a47a5f-8bf7-4260-9ab7-a00caf9ee0fb_template_structured.md', 'utf-8')
};
const activeChapter = "一、投标函";
const cleanActiveChapter = activeChapter.replace(/^第[一二三四五六七八九十百]+章/g, '').trim();

const lines = data.original_markdown.split('\n');
let capturing = false;
let captureLevel = 0;
const result = [];

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    const headingMatch = line.match(/^(#{1,6})\s+(.*)/);
    
    if (headingMatch) {
        const level = headingMatch[1].length;
        const title = headingMatch[2].trim();
        
        if (!capturing) {
            if (title.includes(activeChapter) || (cleanActiveChapter && title.includes(cleanActiveChapter))) {
                capturing = true;
                captureLevel = level;
                result.push(line);
            }
        } else {
            if (level <= captureLevel) {
                break;
            } else {
                result.push(line);
            }
        }
    } else if (capturing) {
        result.push(line);
    }
}

console.log("length:", result.length);
console.log(result.slice(0, 10).join("\n"));
